include/boost/url/detail/impl/any_query_iter.ipp

```diff
@@ -217,17 +217,15 @@ copy_impl(
 {
     (void)end;
     // avoid self-copy
+    std::size_t n = key.size();
+    BOOST_ASSERT(end - n >= dest);
     if( key.data() != dest &&
         key.data() != nullptr)
     {
-        std::size_t n =
```

Comment:
This snippet is here for correctness in case we ever use the iterator differently, but it's [impossible to cover](https://codecov.io/github/CPPAlliance/url/commit/2804061fa80425fd6f5e672f3f3663793d616fa3) this line without indirectly through `replace_value`. Is there anything, in particular, we should do here?

Reply:
if you remove the snippet here you should also remove it in the other places. you could just assert that data() is not null.